src/Main.elm

```diff
@@ -1,10 +1,571 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience exposing (Audience, AudienceType(..), audiencesJSON)
+import Data.AudienceFolder exposing (AudienceFolder, audienceFoldersJSON)
+import Dict exposing (Dict)
+import FontAwesome as FA
+import Html exposing (..)
+import Html.Attributes exposing (class, classList, id)
+import Html.Events exposing (onClick)
+import Json.Decode as Decode exposing (Decoder, decodeString, field, float, int, list, nullable, string)
+import Json.Decode.Pipeline exposing (optional, required, requiredAt)
+import Tree as Tree exposing (Tree)
+import Tree.Zipper as Zipper
 
 
-main : Html msg
+type Node
+    = Dir Id AudienceFolder
+    | File Id Audience
+    | Root
+
+
+type alias Id =
+    Int
```

Comment:
Both `Audience`s and `AudienceFolder`s seem to share the same `Id` space in your implementation, although they have separate `Id` spaces in the database (each has its own table with its own autoincrement sequence).

What alternatives to the above types would you suggest if we wanted to keep the audience and audience folder IDs separate and non-switchable?

Reply:
How about `type alias AudienceId = Int` and `type alias AudienceFolderId = Int` ? It's still somewhat switchable parameter-wise. We can do `type AudienceId = AudienceId Int` to be more explicit in the code.

Reply:
I settled went with
```elm
type Node
    = Dir AudienceFolderId AudienceFolder
    | File AudienceId Audience
    | Root


type alias Id =
    Int


type AudienceId
    = AudienceId Id


type AudienceFolderId
    = AudienceFolderId Id


type RootId
    = RootId Id


type NodeId
    = NodeDir AudienceFolderId
    | NodeFile AudienceId
    | NodeRoot RootId
```
The reason for additional `NodeId` so I can have a generic function like `nodeIdToNode : NodeId -> Node`